From 4734741d33701b6603ec1f1fcf54173556fdf2fe Mon Sep 17 00:00:00 2001
From: "Mazlan, Hazwan Arif" <hazwan.arif.mazlan@intel.com>
Date: Mon, 25 Nov 2024 14:31:29 +0800
Subject: [PATCH 08/11] iris: Added BMG PO Device ID

Signed-off-by: Mazlan, Hazwan Arif <hazwan.arif.mazlan@intel.com>
---
 include/pci_ids/iris_pci_ids.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/pci_ids/iris_pci_ids.h b/include/pci_ids/iris_pci_ids.h
index c81740c9722..cd1b5753852 100644
--- a/include/pci_ids/iris_pci_ids.h
+++ b/include/pci_ids/iris_pci_ids.h
@@ -277,7 +277,9 @@ CHIPSET(0xe202, bmg, "BMG G21", "Intel(R) Graphics")
 CHIPSET(0xe20b, bmg, "BMG G21", "Intel(R) Graphics")
 CHIPSET(0xe20c, bmg, "BMG G21", "Intel(R) Graphics")
 CHIPSET(0xe20d, bmg, "BMG G21", "Intel(R) Graphics")
+CHIPSET(0xe211, bmg, "BMG G21", "Intel(R) Arc(tm) Pro B60 Graphics")
 CHIPSET(0xe212, bmg, "BMG G21", "Intel(R) Graphics")
+CHIPSET(0xe216, bmg, "BMG NEX2", "Intel(R) Graphics")
 
 CHIPSET(0xb080, ptl, "PTL", "Intel(R) Graphics", FORCE_PROBE)
 CHIPSET(0xb081, ptl, "PTL", "Intel(R) Graphics", FORCE_PROBE)
-- 
2.34.1

