From 4fe28582188873b5487a076ebf2e13b3aeb8b6a5 Mon Sep 17 00:00:00 2001
From: Jordan Justen <jordan.l.justen@intel.com>
Date: Tue, 10 Oct 2023 00:31:44 -0700
Subject: [PATCH 13/15] intel/dev: Add WCL device info

Signed-off-by: Jordan Justen <jordan.l.justen@intel.com>
---
 src/intel/dev/intel_device_info.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/intel/dev/intel_device_info.c b/src/intel/dev/intel_device_info.c
index 3cfa68d813d..7c8337abe69 100644
--- a/src/intel/dev/intel_device_info.c
+++ b/src/intel/dev/intel_device_info.c
@@ -1273,6 +1273,12 @@ static const struct intel_device_info intel_device_info_ptl = {
    .has_local_mem = false,
 };
 
+UNUSED static const struct intel_device_info intel_device_info_wcl = {
+   XE3_FEATURES,
+   .platform = INTEL_PLATFORM_WCL,
+   .has_local_mem = false,
+};
+
 void
 intel_device_info_topology_reset_masks(struct intel_device_info *devinfo)
 {
-- 
2.34.1

